sql/character-set-support.md

```diff
@@ -0,0 +1,202 @@
+_cs | 大小写敏感
+_bin | 二进制（Binary）
+
+**注意**：目前为止 TiDB 只支持部分以上提到的 Collation。
+
+## 数据库 Character Set 和 Collation
+
+每个数据库都有相应的 Character Set 和 Collation，`CREATE DATABASE` 可以指定数据库的字符集和排序规则：
+
+```
+CREATE DATABASE db_name
+    [[DEFAULT] CHARACTER SET charset_name]
+    [[DEFAULT] COLLATE collation_name]
+```
+
+在这里 `SCHEMA` 可以跟 `DATABASE` 互换使用。
+
+不同的数据库之间可以使用不一样的字符集和排序规则。
+
+系统变量 `character_set_database` 和 `collation_database` 可以查看到当前数据库的字符集以及排序规则：
+
+```
+mysql> create schema test1 character set utf8 COLLATE uft8_general_ci;
+Query OK, 0 rows affected (0.09 sec)
+
+mysql> use test1;
+Database changed
+mysql> SELECT @@character_set_database, @@collation_database;
++--------------------------+----------------------+
+| @@character_set_database | @@collation_database |
++--------------------------+----------------------+
+| utf8                     | uft8_general_ci      |
++--------------------------+----------------------+
+1 row in set (0.00 sec)
+
+mysql> create schema test2 character set latin1 COLLATE latin1_general_ci;
+Query OK, 0 rows affected (0.09 sec)
+
+mysql> use test2;
+Database changed
+mysql> SELECT @@character_set_database, @@collation_database;
++--------------------------+----------------------+
+| @@character_set_database | @@collation_database |
++--------------------------+----------------------+
+| latin1                   | latin1_general_ci    |
++--------------------------+----------------------+
+1 row in set (0.00 sec)
+```
+
+在 INFORMATION_SCHEMA 中也可以查看到这两个值：
+
+```
+SELECT DEFAULT_CHARACTER_SET_NAME, DEFAULT_COLLATION_NAME
+FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = 'db_name';
+```
+
+## 表的 Character Set 和 Collation
+
+表的 Character Set 和 Collation 可以通过以下语句来设置：
+
+```
+CREATE TABLE tbl_name (column_list)
+    [[DEFAULT] CHARACTER SET charset_name]
+    [COLLATE collation_name]]
+
+ALTER TABLE tbl_name
+    [[DEFAULT] CHARACTER SET charset_name]
+    [COLLATE collation_name]
+```
+
+例如：
+
+```
+mysql> CREATE TABLE t1(a int) CHARACTER SET utf8 COLLATE utf8_general_ci;
+Query OK, 0 rows affected (0.08 sec)
+```
+
+如果 Column 的字符集和排序规则没有设置，那么表的字符集和排序规则就作为其默认值。
+
+## 列的 Character Set 和 Collation
+
+列的 Character Set 和 Collation 的语法如下：
+
+```
+col_name {CHAR | VARCHAR | TEXT} (col_length)
+    [CHARACTER SET charset_name]
+    [COLLATE collation_name]
+
+col_name {ENUM | SET} (val_list)
+    [CHARACTER SET charset_name]
+    [COLLATE collation_name]
+```
+
+## 客户端连接的 Character Sets 和 Collations
+
+* 服务器的字符集和排序规则可以通过系统变量 `character_set_server` 和 `collation_server` 
+* 数据库的字符集和排序规则可以通过环境变量 ` character_set_database` 和 `collation_database`
+
+对于每一个客户端的连接，也有相应的变量表示字符集和排序规则：`character_set_connection` 和 `collation_connection`
+
+`character_set_client` 代表客户端的字符集。 在返回结果前，服务端会把结果根据 `character_set_results` 转换成对应的字符集。包括结果的元信息等。
+
+可以用以下的语句来影响这些跟客户端相关的字符集变量：
+
+* `SET NAMES 'charset_name' [COLLATE 'collation_name']` 
+    `SET NAMES` 用来设定客户端会在之后的请求中使用的字符集。`SET NAMES utf8` 表示客户端会在接下来的请求中，都使用 utf8 字符集。服务端也会在之后返回结果的时候使用 utf8 字符集。
```

Comment:
Indent style is wrong. Markdown has no definition list. Using paragraph or embedded list is ok